NPE when sub_error is null in AAD response

[ADO: Tracking (corpnet reqd)](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1063189)

----
Affects: MSAL 1.5.3

Root cause:
Querying `sub_error` with `.equals()` prior to asserting it is not `null` -- `sub_error` is _optional_.

```java
Attempt: 0, Failed to refresh tokens No authentication result was received. An exception occurred: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference com.microsoft.identity.client.exception.MsalClientException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
    at com.microsoft.identity.client.internal.controllers.MsalExceptionAdapter.msalExceptionFromBaseException(MsalExceptionAdapter.java:51)
    at com.microsoft.identity.client.PublicClientApplication$16.onError(PublicClientApplication.java:1673)
    at com.microsoft.identity.client.PublicClientApplication$16.onError(PublicClientApplication.java:1664)
    at com.microsoft.identity.common.internal.controllers.CommandDispatcher$3.run(CommandDispatcher.java:209)
    at android.os.Handler.handleCallback(Handler.java:873)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:224)
    at android.app.ActivityThread.main(ActivityThread.java:7110)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:536)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:876)
Caused by: com.microsoft.identity.common.exception.ClientException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
    at com.microsoft.identity.common.internal.controllers.ExceptionAdapter.baseExceptionFromException(ExceptionAdapter.java:254)
    at com.microsoft.identity.common.internal.controllers.CommandDispatcher.executeCommand(CommandDispatcher.java:171)
    at com.microsoft.identity.common.internal.controllers.CommandDispatcher.access$200(CommandDispatcher.java:65)
    at com.microsoft.identity.common.internal.controllers.CommandDispatcher$2.run(CommandDispatcher.java:126)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
    at com.microsoft.identity.common.internal.controllers.BaseController.renewAccessToken(BaseController.java:311)
    at com.microsoft.identity.client.internal.controllers.LocalMSALController.acquireTokenSilent(LocalMSALController.java:309)
    at com.microsoft.identity.common.internal.commands.SilentTokenCommand.execute(SilentTokenCommand.java:75)
    at com.microsoft.identity.common.internal.commands.SilentTokenCommand.execute(SilentTokenCommand.java:41)
    at com.microsoft.identity.common.internal.controllers.CommandDispatcher.executeCommand(CommandDispatcher.java:166)
```

--- a/common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java
+++ b/common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java
@@ -25,6 +25,9 @@ package com.microsoft.identity.common.internal.controllers;
 import android.content.Intent;
 import android.text.TextUtils;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
 import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
 import com.microsoft.identity.common.adal.internal.net.HttpWebRequest;
 import com.microsoft.identity.common.adal.internal.util.StringExtensions;
@@ -81,9 +84,6 @@ import java.util.List;
 import java.util.Set;
 import java.util.UUID;
 import java.util.regex.Pattern;
-
-import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
 
 import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.OAuth2ErrorCode.INVALID_GRANT;
 import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.OAuth2SubErrorCode.BAD_TOKEN;
@@ -314,7 +314,7 @@ public abstract class BaseController {
                             "attempting to delete the RT from cache, result:" + isRemoved
                     );
                 }
-            }else {
+            } else {
                 Logger.warn(TAG, "Invalid state, No token success or error response on the token result");
             }
         }
@@ -581,7 +581,7 @@ public abstract class BaseController {
 
     @Nullable
     private AccountRecord getAccountWithFRTIfAvailable(@NonNull final SilentTokenCommandParameters parameters,
-                                                       @NonNull final MsalOAuth2TokenCache msalOAuth2TokenCache){
+                                                       @NonNull final MsalOAuth2TokenCache msalOAuth2TokenCache) {
 
         final String homeAccountId = parameters.getAccount().getHomeAccountId();
         final String clientId = parameters.getClientId();

--- a/common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java
+++ b/common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java
@@ -308,10 +308,14 @@ public abstract class BaseController {
                 final String subErrorCode = tokenResult.getErrorResponse().getSubError();
                 Logger.info(TAG, "Error: " + errorCode + " Suberror: " + subErrorCode);
 
-                if (errorCode.equals(INVALID_GRANT) && subErrorCode.equals(BAD_TOKEN)) {
+                if (null != errorCode && errorCode.equals(INVALID_GRANT)
+                        && null != subErrorCode && subErrorCode.equals(BAD_TOKEN)) {
                     boolean isRemoved = tokenCache.removeCredential(cacheRecord.getRefreshToken());
-                    Logger.info(TAG, "Refresh token is invalid, " +
-                            "attempting to delete the RT from cache, result:" + isRemoved
+                    Logger.info(
+                            TAG,
+                            "Refresh token is invalid, "
+                                    + "attempting to delete the RT from cache, result:"
+                                    + isRemoved
                     );
                 }
             } else {

--- a/common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java
+++ b/common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java
@@ -308,8 +308,7 @@ public abstract class BaseController {
                 final String subErrorCode = tokenResult.getErrorResponse().getSubError();
                 Logger.info(TAG, "Error: " + errorCode + " Suberror: " + subErrorCode);
 
-                if (null != errorCode && errorCode.equals(INVALID_GRANT)
-                        && null != subErrorCode && subErrorCode.equals(BAD_TOKEN)) {
+                if (INVALID_GRANT.equals(errorCode) && BAD_TOKEN.equals(subErrorCode)) {
                     boolean isRemoved = tokenCache.removeCredential(cacheRecord.getRefreshToken());
                     Logger.info(
                             TAG,

